cm2350/gdbdefs/e200z759n3.py

```diff
@@ -0,0 +1,423 @@
+    ('IVOR8', (32, 114)),
+    ('IVOR9', (32, 115)),
+    ('IVOR10', (32, 116)),
+    ('IVOR11', (32, 117)),
+    ('IVOR12', (32, 118)),
+    ('IVOR13', (32, 119)),
+    ('IVOR14', (32, 120)),
+    ('IVOR15', (32, 121)),
+    ('IVOR38', (32, 122)),
+    ('IVOR39', (32, 123)),
+    ('IVOR40', (32, 124)),
+    ('IVOR41', (32, 125)),
+    ('IVOR42', (32, 126)),
+    ('TENSR', (32, 127)),
+    ('TENS', (32, 128)),
+    ('TENC', (32, 129)),
+    ('TIR', (32, 130)),
+    ('L1CFG0', (32, 131)),
+    ('L1CFG1', (32, 132)),
+    ('NPIDR5', (32, 133)),
+    ('L2CFG0', (32, 134)),
+    ('IVOR32', (32, 135)),
+    ('IVOR33', (32, 136)),
+    ('IVOR34', (32, 137)),
+    ('IVOR35', (32, 138)),
+    ('IVOR36', (32, 139)),
+    ('IVOR37', (32, 140)),
+    ('DBCR3', (32, 141)),
+    ('DBCNT', (32, 142)),
+    ('DBCR4', (32, 143)),
+    ('DBCR5', (32, 144)),
+    ('MCARU', (32, 145)),
+    ('MCSRR0', (32, 146)),
+    ('MCSRR1', (32, 147)),
+    ('MCSR', (32, 148)),
+    ('MCAR', (32, 149)),
+    ('DSRR0', (32, 150)),
+    ('DSRR1', (32, 151)),
+    ('DDAM', (32, 152)),
+    ('L1CSR2', (32, 153)),
+    ('L1CSR3', (32, 154)),
+    ('MAS0', (32, 155)),
+    ('MAS1', (32, 156)),
+    ('MAS2', (32, 157)),
+    ('MAS3', (32, 158)),
+    ('MAS4', (32, 159)),
+    ('MAS6', (32, 160)),
+    ('PID1', (32, 161)),
+    ('PID2', (32, 162)),
+    ('EDBRAC0', (32, 163)),
+    ('TLB0CFG', (32, 164)),
+    ('TLB1CFG', (32, 165)),
+    ('TLB2CFG', (32, 166)),
+    ('TLB3CFG', (32, 167)),
+    ('DBRR0', (32, 168)),
+    ('EPR', (32, 169)),
+    ('L2ERRINTEN', (32, 170)),
+    ('L2ERRATTR', (32, 171)),
+    ('L2ERRADDR', (32, 172)),
+    ('L2ERREADDR', (32, 173)),
+    ('L2ERRCTL', (32, 174)),
+    ('L2ERRDIS', (32, 175)),
+    ('L1FINV1', (32, 176)),
+    ('DEVENT', (32, 177)),
+    ('NSPD', (32, 178)),
+    ('NSPC', (32, 179)),
+    ('L2ERRINJHI', (32, 180)),
+    ('L2ERRINJLO', (32, 181)),
+    ('L2ERRINJCTL', (32, 182)),
+    ('L2CAPTDATAHI', (32, 183)),
+    ('L2CAPTDATALO', (32, 184)),
+    ('L2CAPTECC', (32, 185)),
+    ('L2ERRDET', (32, 186)),
+    ('HID0', (32, 187)),
+    ('HID1', (32, 188)),
+    ('L1CSR0', (32, 189)),
+    ('L1CSR1', (32, 190)),
+    ('MMUCSR0', (32, 191)),
+    ('BUCSR', (32, 192)),
+    ('MMUCFG', (32, 193)),
+    ('L1FINV0', (32, 194)),
+    ('L2CSR0', (32, 195)),
+    ('L2CSR1', (32, 196)),
+    ('PWRMGTCR0', (32, 197)),
+    ('SCCSRBAR', (32, 198)),
+    ('SVR', (32, 199)),
+])
+
+# r0 through xer, but the size is the size of the asciified GDB formatted 
+# packet.
+# 32 GPRs, 6 system registers = 38 registers
+# All registers for this interface are 32bits wide, 4 bytes = 8 characters
+reg_pkt_size = (38 * 8, 38)
+
+target_xml = b'''<?xml version="1.0"?>
+<!DOCTYPE target SYSTEM "gdb-target.dtd">
+<target>
+  <architecture>powerpc:vle</architecture>
+  <feature name="org.gnu.gdb.power.core">
+    <reg bitsize="32" name="r0" regnum="0" />
+    <reg bitsize="32" name="r1" regnum="1" />
+    <reg bitsize="32" name="r2" regnum="2" />
```

Comment:
This looks like it came from the PEMicro server (or some other GDB server).  What's the story, and why not generate it?

Reply:
Similar answer to the previous comment, things almost but not quite line up with the way we defined the register groups. And it appears GDB expects things to match _exactly_. I couldn't come up with a generic way to handle the discrepancies.